### PR TITLE
`ls` implement size display

### PR DESCRIPTION
**Pull request by hbina**
Implements the flag `-s` or `--size` to display the size of file in blocks.
Depends on for the refactor https://github.com/uutils/coreutils/pull/2436

### CODE REVIEW

**Comment by sylvestre** [comment]
Please add a test for this :)
thanks

**Comment by hbina** [comment]
Added some tests based on this

```bash
coreutils on  hbina-ls-refactor-options-module [!?] is 📦 v0.0.6 via 🦀 v1.53.0 took 3s 
❯ for i in {0..123456..1}; do echo "a"; done > number.txt && ls -s number.txt 
244 number.txt

coreutils on  hbina-ls-refactor-options-module [!?] is 📦 v0.0.6 via 🦀 v1.53.0 took 3s 
❯ for i in {0..234567..1}; do echo "a"; done > number.txt && ls -s number.txt 
460 number.txt

coreutils on  hbina-ls-refactor-options-module [!?] is 📦 v0.0.6 via 🦀 v1.53.0 took 7s 
❯ for i in {0..345678..1}; do echo "a"; done > number.txt && ls -s number.txt 
676 number.txt

coreutils on  hbina-ls-refactor-options-module [!?] is 📦 v0.0.6 via 🦀 v1.53.0 took 10s 
❯ for i in {0..456789..1}; do echo "a"; done > number.txt && ls -s number.txt 
896 number.txt
```

File: tests/by-util/test_ls.rs [hunk 1]
```diff
@@ -2021,3 +2021,38 @@ fn test_ls_path() {
...
+
+#[test]
+fn test_ls_size() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.touch("test-1");
+    at.append(
+        "test-1",
+        &std::iter::repeat("a\n").take(123456).collect::<String>(),
```
Comment (sylvestre) [inline tests/by-util/test_ls.rs:2033]: clippy isn't happy here :)

**Comment by sylvestre** [comment]
This test fails:
```
---- test_ls::test_ls_size stdout ----
current_directory_resolved: 
touch: C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-1
open(append): C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-1
touch: C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-2
open(append): C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-2
touch: C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-3
open(append): C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-3
touch: C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-4
open(append): C:\Users\RUNNER~1\AppData\Local\Temp\.tmpJKz62i\test-4
run: D:\a\coreutils\coreutils\target\i686-pc-windows-gnu\debug\coreutils.exe ls -s
run: D:\a\coreutils\coreutils\target\i686-pc-windows-gnu\debug\coreutils.exe ls -s
thread 'test_ls::test_ls_size' panicked at 'assertion failed: `(left == right)`

Diff < left / right > :
<"246912 test-1  469134 test-2  691356 test-3  913578 test-4\n"
>"test-4  test-3  test-2  test-1\n"

', tests\common\util.rs:222:9
```

File: tests/by-util/test_ls.rs (hunk 1 above)
Comment (hbina) [reply tests/by-util/test_ls.rs:2033]: I don't see it?

```
hbina@DESKTOP-2D0I5MJ:~/git/coreutils$ cargo clippy --all-targets -- --cfg true
   Compiling coreutils v0.0.6 (/home/hbina/git/coreutils)
   Compiling uu_factor v0.0.6 (/home/hbina/git/coreutils/src/uu/factor)
   Compiling uucore_procs v0.0.5 (/home/hbina/git/coreutils/src/uucore_procs)
    Checking uucore v0.0.8 (/home/hbina/git/coreutils/src/uucore)
    Checking uu_base32 v0.0.6 (/home/hbina/git/coreutils/src/uu/base32)
    Checking uu_printf v0.0.6 (/home/hbina/git/coreutils/src/uu/printf)
    Checking uu_mktemp v0.0.6 (/home/hbina/git/coreutils/src/uu/mktemp)
    Checking uu_shuf v0.0.6 (/home/hbina/git/coreutils/src/uu/shuf)
    Checking uu_env v0.0.6 (/home/hbina/git/coreutils/src/uu/env)
    Checking uu_printenv v0.0.6 (/home/hbina/git/coreutils/src/uu/printenv)
    Checking uu_csplit v0.0.6 (/home/hbina/git/coreutils/src/uu/csplit)
    Checking uu_cp v0.0.6 (/home/hbina/git/coreutils/src/uu/cp)
    Checking uu_head v0.0.6 (/home/hbina/git/coreutils/src/uu/head)
    Checking uu_tail v0.0.6 (/home/hbina/git/coreutils/src/uu/tail)
    Checking uu_join v0.0.6 (/home/hbina/git/coreutils/src/uu/join)
    Checking uu_df v0.0.6 (/home/hbina/git/coreutils/src/uu/df)
    Checking uu_cut v0.0.6 (/home/hbina/git/coreutils/src/uu/cut)
    Checking uu_mkdir v0.0.6 (/home/hbina/git/coreutils/src/uu/mkdir)
    Checking uu_dirname v0.0.6 (/home/hbina/git/coreutils/src/uu/dirname)
    Checking uu_du v0.0.6 (/home/hbina/git/coreutils/src/uu/du)
    Checking uu_echo v0.0.6 (/home/hbina/git/coreutils/src/uu/echo)
    Checking uu_split v0.0.6 (/home/hbina/git/coreutils/src/uu/split)
    Checking uu_tsort v0.0.6 (/home/hbina/git/coreutils/src/uu/tsort)
    Checking uu_paste v0.0.6 (/home/hbina/git/coreutils/src/uu/paste)
    Checking uu_sum v0.0.6 (/home/hbina/git/coreutils/src/uu/sum)
    Checking uu_relpath v0.0.6 (/home/hbina/git/coreutils/src/uu/relpath)
    Checking uu_ptx v0.0.6 (/home/hbina/git/coreutils/src/uu/ptx)
    Checking uu_truncate v0.0.6 (/home/hbina/git/coreutils/src/uu/truncate)
    Checking uu_pwd v0.0.6 (/home/hbina/git/coreutils/src/uu/pwd)
    Checking uu_mv v0.0.6 (/home/hbina/git/coreutils/src/uu/mv)
    Checking uu_fmt v0.0.6 (/home/hbina/git/coreutils/src/uu/fmt)
    Checking uu_dircolors v0.0.6 (/home/hbina/git/coreutils/src/uu/dircolors)
    Checking uu_uniq v0.0.6 (/home/hbina/git/coreutils/src/uu/uniq)
    Checking uu_ln v0.0.6 (/home/hbina/git/coreutils/src/uu/ln)
    Checking uu_more v0.0.6 (/home/hbina/git/coreutils/src/uu/more)
    Checking uu_test v0.0.6 (/home/hbina/git/coreutils/src/uu/test)
    Checking uu_ls v0.0.6 (/home/hbina/git/coreutils/src/uu/ls)
    Checking uu_cat v0.0.6 (/home/hbina/git/coreutils/src/uu/cat)
    Checking uu_cksum v0.0.6 (/home/hbina/git/coreutils/src/uu/cksum)
    Checking uu_seq v0.0.6 (/home/hbina/git/coreutils/src/uu/seq)
    Checking uu_false v0.0.6 (/home/hbina/git/coreutils/src/uu/false)
    Checking uu_expand v0.0.6 (/home/hbina/git/coreutils/src/uu/expand)
    Checking uu_rmdir v0.0.6 (/home/hbina/git/coreutils/src/uu/rmdir)
    Checking uu_comm v0.0.6 (/home/hbina/git/coreutils/src/uu/comm)
    Checking uu_date v0.0.6 (/home/hbina/git/coreutils/src/uu/date)
    Checking uu_realpath v0.0.6 (/home/hbina/git/coreutils/src/uu/realpath)
    Checking uu_shred v0.0.6 (/home/hbina/git/coreutils/src/uu/shred)
    Checking uu_sleep v0.0.6 (/home/hbina/git/coreutils/src/uu/sleep)
    Checking uu_wc v0.0.6 (/home/hbina/git/coreutils/src/uu/wc)
    Checking uu_true v0.0.6 (/home/hbina/git/coreutils/src/uu/true)
    Checking uu_basename v0.0.6 (/home/hbina/git/coreutils/src/uu/basename)
    Checking uu_fold v0.0.6 (/home/hbina/git/coreutils/src/uu/fold)
    Checking uu_link v0.0.6 (/home/hbina/git/coreutils/src/uu/link)
    Checking uu_rm v0.0.6 (/home/hbina/git/coreutils/src/uu/rm)
    Checking uu_expr v0.0.6 (/home/hbina/git/coreutils/src/uu/expr)
    Checking uu_yes v0.0.6 (/home/hbina/git/coreutils/src/uu/yes)
    Checking uu_tac v0.0.6 (/home/hbina/git/coreutils/src/uu/tac)
    Checking uu_readlink v0.0.6 (/home/hbina/git/coreutils/src/uu/readlink)
    Checking uu_nl v0.0.6 (/home/hbina/git/coreutils/src/uu/nl)
    Checking uu_tee v0.0.6 (/home/hbina/git/coreutils/src/uu/tee)
    Checking uu_hashsum v0.0.6 (/home/hbina/git/coreutils/src/uu/hashsum)
    Checking uu_unexpand v0.0.6 (/home/hbina/git/coreutils/src/uu/unexpand)
    Checking uu_pr v0.0.6 (/home/hbina/git/coreutils/src/uu/pr)
    Checking uu_sort v0.0.6 (/home/hbina/git/coreutils/src/uu/sort)
    Checking uu_od v0.0.6 (/home/hbina/git/coreutils/src/uu/od)
    Checking uu_tr v0.0.6 (/home/hbina/git/coreutils/src/uu/tr)
    Checking uu_base64 v0.0.6 (/home/hbina/git/coreutils/src/uu/base64)
    Finished dev [unoptimized + debuginfo] target(s) in 7.23s
```

Comment (miDeb) [reply tests/by-util/test_ls.rs:2033]: Running `cargo clippy --all-targets` should do it.

Comment (hbina) [reply tests/by-util/test_ls.rs:2033]: Nothing

```
coreutils on  hbina-ls-implement-size-display [?] is 📦 v0.0.6 via 🦀 v1.53.0 took 2s 
❯ cargo clippy --all-targets
   Compiling coreutils v0.0.6 (/home/hbina/git/coreutils)
    Checking rlimit v0.4.0
    Checking users v0.10.0
   Compiling uu_factor v0.0.6 (/home/hbina/git/coreutils/src/uu/factor)
   Compiling uucore_procs v0.0.5 (/home/hbina/git/coreutils/src/uucore_procs)
    Checking uucore v0.0.8 (/home/hbina/git/coreutils/src/uucore)
    Checking uu_base32 v0.0.6 (/home/hbina/git/coreutils/src/uu/base32)
    Checking uu_od v0.0.6 (/home/hbina/git/coreutils/src/uu/od)
    Checking uu_truncate v0.0.6 (/home/hbina/git/coreutils/src/uu/truncate)
    Checking uu_mktemp v0.0.6 (/home/hbina/git/coreutils/src/uu/mktemp)
    Checking uu_rm v0.0.6 (/home/hbina/git/coreutils/src/uu/rm)
    Checking uu_readlink v0.0.6 (/home/hbina/git/coreutils/src/uu/readlink)
    Checking uu_false v0.0.6 (/home/hbina/git/coreutils/src/uu/false)
    Checking uu_fold v0.0.6 (/home/hbina/git/coreutils/src/uu/fold)
    Checking uu_tail v0.0.6 (/home/hbina/git/coreutils/src/uu/tail)
    Checking uu_tac v0.0.6 (/home/hbina/git/coreutils/src/uu/tac)
    Checking uu_hashsum v0.0.6 (/home/hbina/git/coreutils/src/uu/hashsum)
    Checking uu_seq v0.0.6 (/home/hbina/git/coreutils/src/uu/seq)
    Checking uu_date v0.0.6 (/home/hbina/git/coreutils/src/uu/date)
    Checking uu_test v0.0.6 (/home/hbina/git/coreutils/src/uu/test)
    Checking uu_expr v0.0.6 (/home/hbina/git/coreutils/src/uu/expr)
    Checking uu_csplit v0.0.6 (/home/hbina/git/coreutils/src/uu/csplit)
    Checking uu_link v0.0.6 (/home/hbina/git/coreutils/src/uu/link)
    Checking uu_pr v0.0.6 (/home/hbina/git/coreutils/src/uu/pr)
    Checking uu_head v0.0.6 (/home/hbina/git/coreutils/src/uu/head)
    Checking uu_paste v0.0.6 (/home/hbina/git/coreutils/src/uu/paste)
    Checking uu_realpath v0.0.6 (/home/hbina/git/coreutils/src/uu/realpath)
    Checking uu_nl v0.0.6 (/home/hbina/git/coreutils/src/uu/nl)
    Checking uu_unexpand v0.0.6 (/home/hbina/git/coreutils/src/uu/unexpand)
    Checking uu_df v0.0.6 (/home/hbina/git/coreutils/src/uu/df)
    Checking uu_printenv v0.0.6 (/home/hbina/git/coreutils/src/uu/printenv)
    Checking uu_yes v0.0.6 (/home/hbina/git/coreutils/src/uu/yes)
    Checking uu_sort v0.0.6 (/home/hbina/git/coreutils/src/uu/sort)
    Checking uu_ls v0.0.6 (/home/hbina/git/coreutils/src/uu/ls)
    Checking uu_more v0.0.6 (/home/hbina/git/coreutils/src/uu/more)
    Checking uu_du v0.0.6 (/home/hbina/git/coreutils/src/uu/du)
    Checking uu_cp v0.0.6 (/home/hbina/git/coreutils/src/uu/cp)
    Checking uu_dirname v0.0.6 (/home/hbina/git/coreutils/src/uu/dirname)
    Checking uu_printf v0.0.6 (/home/hbina/git/coreutils/src/uu/printf)
    Checking uu_mv v0.0.6 (/home/hbina/git/coreutils/src/uu/mv)
    Checking uu_cat v0.0.6 (/home/hbina/git/coreutils/src/uu/cat)
    Checking uu_sum v0.0.6 (/home/hbina/git/coreutils/src/uu/sum)
    Checking uu_fmt v0.0.6 (/home/hbina/git/coreutils/src/uu/fmt)
    Checking uu_tsort v0.0.6 (/home/hbina/git/coreutils/src/uu/tsort)
    Checking uu_expand v0.0.6 (/home/hbina/git/coreutils/src/uu/expand)
    Checking uu_shuf v0.0.6 (/home/hbina/git/coreutils/src/uu/shuf)
    Checking uu_join v0.0.6 (/home/hbina/git/coreutils/src/uu/join)
    Checking uu_env v0.0.6 (/home/hbina/git/coreutils/src/uu/env)
    Checking uu_tee v0.0.6 (/home/hbina/git/coreutils/src/uu/tee)
    Checking uu_true v0.0.6 (/home/hbina/git/coreutils/src/uu/true)
    Checking uu_shred v0.0.6 (/home/hbina/git/coreutils/src/uu/shred)
    Checking uu_sleep v0.0.6 (/home/hbina/git/coreutils/src/uu/sleep)
    Checking uu_ln v0.0.6 (/home/hbina/git/coreutils/src/uu/ln)
    Checking uu_relpath v0.0.6 (/home/hbina/git/coreutils/src/uu/relpath)
    Checking uu_mkdir v0.0.6 (/home/hbina/git/coreutils/src/uu/mkdir)
    Checking uu_comm v0.0.6 (/home/hbina/git/coreutils/src/uu/comm)
    Checking uu_echo v0.0.6 (/home/hbina/git/coreutils/src/uu/echo)
    Checking uu_cut v0.0.6 (/home/hbina/git/coreutils/src/uu/cut)
    Checking uu_cksum v0.0.6 (/home/hbina/git/coreutils/src/uu/cksum)
    Checking uu_pwd v0.0.6 (/home/hbina/git/coreutils/src/uu/pwd)
    Checking uu_wc v0.0.6 (/home/hbina/git/coreutils/src/uu/wc)
    Checking uu_ptx v0.0.6 (/home/hbina/git/coreutils/src/uu/ptx)
    Checking uu_split v0.0.6 (/home/hbina/git/coreutils/src/uu/split)
    Checking uu_uniq v0.0.6 (/home/hbina/git/coreutils/src/uu/uniq)
    Checking uu_basename v0.0.6 (/home/hbina/git/coreutils/src/uu/basename)
    Checking uu_tr v0.0.6 (/home/hbina/git/coreutils/src/uu/tr)
    Checking uu_rmdir v0.0.6 (/home/hbina/git/coreutils/src/uu/rmdir)
    Checking uu_dircolors v0.0.6 (/home/hbina/git/coreutils/src/uu/dircolors)
    Checking uu_base64 v0.0.6 (/home/hbina/git/coreutils/src/uu/base64)
    Finished dev [unoptimized + debuginfo] target(s) in 8.52s
```

Comment (hbina) [reply tests/by-util/test_ls.rs:2033]: nvm, its in nightly

**Comment by sylvestre** [comment]
Freebsd is failing on

```

---- test_ls::test_ls_size stdout ----
current_directory_resolved: 
touch: /tmp/.tmpc8bXmN/test-1
open(append): /tmp/.tmpc8bXmN/test-1
touch: /tmp/.tmpc8bXmN/test-2
open(append): /tmp/.tmpc8bXmN/test-2
touch: /tmp/.tmpc8bXmN/test-3
open(append): /tmp/.tmpc8bXmN/test-3
touch: /tmp/.tmpc8bXmN/test-4
open(append): /tmp/.tmpc8bXmN/test-4
run: /tmp/cirrus-ci-build/target/debug/coreutils ls -s
run: /tmp/cirrus-ci-build/target/debug/coreutils ls -s
thread 'test_ls::test_ls_size' panicked at 'assertion failed: `(left == right)`

�[1mDiff�[0m �[31m< left�[0m / �[32mright >�[0m :
�[31m<"244 test-1\n�[0m�[1;48;5;52;31m512�[0m�[31m test-2\n7�[0m�[1;48;5;52;31m3�[0m�[31m6 test-3\n�[0m�[1;48;5;52;31m92�[0m�[31m8 test-4\n"�[0m
�[32m>"244 test-1\n�[0m�[1;48;5;22;32m460�[0m�[32m test-2\n�[0m�[1;48;5;22;32m6�[0m�[32m76 test-3\n8�[0m�[1;48;5;22;32m96�[0m�[32m test-4\n"�[0m

', tests/common/util.rs:222:9


failures:
    test_ls::test_ls_size

```

https://github.com/uutils/coreutils/pull/2435/checks?check_run_id=2866873876
could you please fix that?

**Comment by hbina** [comment]
I have no idea how to use FreeBSD. I am trying tho.

**Comment by sylvestre** [comment]
We have a few conflicts on it, sorry

**Comment by hbina** [comment]
Do you have any idea why the fixture doesn't work in FreeBSD, i tried my `ls -la` normally and it seems to work fine.

**Comment by skyronic** [comment]
Hey @hbina 

I'm new to rust and thought I'd implement `ls -s` before I saw that you've got a much more comprehensive implementation here!

Can I ask why you're calculating the block sizes instead of using `md.blocks()` for block sizes? I tried using it and it seemed to work fine in [my basic implementation](https://github.com/skyronic/coreutils/commit/0ca7b91e7a3cf35229c7c596b4aaa4c44f43ea94)

Just curious about why you took this approach, so I can understand it better. Thanks!

**Comment by hbina** [comment]
I actually mixed in some refactoring as well. I think if your implementation works on Windows and BSD, it might be better.

**Comment by skyronic** [comment]
I just realized my implementation doesn't even compile on windows 😅

**Pull request closed without merging**